### PR TITLE
Replace ORKOrderedTaskView with new ResearchKit one

### PR DESCRIPTION
**Pull request by Supereg**
# Replace ORKOrderedTaskView with new ResearchKit one

## :recycle: Current situation & Problem

This PR updates to our latest version of ResearchKit and replaces the internal `ORKOrderedTaskView` implementation with the new one available in `ResearchKitSwiftUI`.

## :gear: Release Notes 
* Updated to new ResearchKit version

## :books: Documentation

--
## :white_check_mark: Testing

--

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/SpeziQuestionnaire/QuestionnaireView.swift
Comment (PSchmiedmayer) [inline Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: I am wondering if we should make this configurable with a default value so we don't make this a breaking change?

**Review by PSchmiedmayer — COMMENTED** [review]
Thank you for the changes. I had one comment that it would be great to discuss before we merge the PR.

File: Sources/SpeziQuestionnaire/QuestionnaireView.swift
Comment (Supereg) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: Not sure what the original motivation was to set this subject field?
In every case you can just easily override or adjust the field when you get the response yourself.
But if it's a required field it might make sense to have it configurable in the initializer, such that it gets visibility.

Comment (vishnuravi) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: @Supereg the subject field has a ["Must Support" flag](https://build.fhir.org/ig/HL7/sdc/StructureDefinition-sdc-questionnaireresponse.html). It is not technically required. I don't think putting in a default value that is not a valid patient identifier makes a lot of sense.

Comment (Supereg) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: > I don't think putting in a default value that is not a valid patient identifier makes a lot of sense.

That's not a new behavior as of this PR. We did that before already. Not sure when this was merged. 


But thanks for the context. Are we fair to assume that we always store a String in there? Or would it technically make sense to support different data types? Otherwise, I would add a required String parameter to the initializer of the Questionnaire View. For backwards compatibility I would maintain the current initializer (providing the current default value) and add a deprecation warning.

Comment (PSchmiedmayer) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: Yes, definitely not your fault @Supereg; I just saw it in the PR diff.

@vishnuravi would you agree that not even writing a value here makes sense? If so, it is up to the user of the framework to set a valid subject identifier?

Comment (vishnuravi) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: According to the specification, the `reference` element of the [Reference](https://hl7.org/fhir/R4/references.html) resource, which is what is stored in the `subject` field is always a string.

Comment (vishnuravi) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: > Yes, definitely not your fault @Supereg; I just saw it in the PR diff.
> 
> @vishnuravi would you agree that not even writing a value here makes sense? If so, it is up to the user of the framework to set a valid subject identifier?

Yes, I would leave it up to the user to set a valid identifier. If they choose not to put any identifier at all, they can, the specification allows it.

Comment (Supereg) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: Alright, so I would basically just remove this line then and leave it to the user to add fields that they might require upon further processing? Or what do you think is the best approach here?

Comment (PSchmiedmayer) [reply Sources/SpeziQuestionnaire/QuestionnaireView.swift:78]: As just discussed in the office: Let's remove it, we don't have a sensible default here and it is up to the user of SpeziQuestionnaire to set this to an appropriate value.

**Review by PSchmiedmayer — APPROVED** [review]

**Comment by Supereg** [comment]
Not sure why codecov doesn't report any updates. Might we just merge?

**Pull request merged**
